portfolio/src/main/java/com/google/sps/servlets/CommentForumServlet.java

```diff
@@ -30,18 +31,30 @@
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
-public class DataServlet extends HttpServlet {
+public class CommentForumServlet extends HttpServlet {
 
+  private static final Query query = new Query("Comment")
+      .addSort("timestamp", SortDirection.DESCENDING);
+  private static final Gson gson = new Gson();
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    String numCommentsToDisplayString = request.getParameter("num-comments-to-display");
+    // Convert the input to an int.
+    int numCommentsToDisplay;
+    try {
+      numCommentsToDisplay = Integer.parseInt(numCommentsToDisplayString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + numCommentsToDisplayString);
+      return;
+    }
+
```

Comment:
It's worth doing server-side validation here. Imaging someone works around the JS and enters '100,000,000,000' as the number of comments. If there are that many comments the resulting response could be so large that it crashes the whole server.